app/app.jsx

```diff
@@ -1,32 +1,27 @@
-import React from 'react';
+import React, {PropTypes} from 'react';
 import MessageList from './components/message-list';
 import MessageBox from './components/message-box';
-import {store} from './store';
-import {Map, List} from 'immutable';
-import {addChannel} from './actions/channels';
-import {addMessage} from './actions/messages';
+import {connect} from 'react-redux';
 import {startSocketClient} from './core/socket';
 
-// Put some dummy data to store
-store.dispatch(addChannel(Map({id: 0, name: '0 channel', userIds: List.of(0) })));
-store.dispatch(addMessage(Map({id: 0, channelId: 0, senderId: 0, text: 'test message'})));
-store.dispatch(addMessage(Map({id: 1, channelId: 0, senderId: 1, text: 'test message1'})));
-store.dispatch(addMessage(Map({id: 2, channelId: 0, senderId: 0, text: 'test message2'})));
-store.dispatch(addMessage(Map({id: 3, channelId: 0, senderId: 1, text: 'test message3'})));
-
-const msgs = store.getState().messages;
 
 startSocketClient();
 
+@connect(state => ({ messages: state.messages }))
```

Comment:
Btw, how does this work without passing component to a second function of the decorator?